cpp/open3d/geometry/PointCloud.cpp

```diff
@@ -158,6 +158,44 @@ std::vector<double> PointCloud::ComputePointCloudDistance(
     return distances;
 }
 
+PointCloud &PointCloud::RemoveDuplicatedPoints() {
+    bool has_normals = HasNormals();
+    bool has_colors = HasColors();
+    bool has_covariances = HasCovariances();
+    size_t old_points_num = points_.size();
+    size_t k = 0;
+
+    typedef std::tuple<double, double, double> Coordinate3;
+    std::unordered_map<Coordinate3, size_t, utility::hash_tuple<Coordinate3>>
```

Comment:
Is there any particular reason to use `unordered_map` instead of `unordered_set` here? 

If you want to return `std::tuple<PointCloud, std::vector<size_t>>&` i.e. 
`pointcloud, indices_of_removed_points = pcd.remove_duplicate_points()`, then we can keep using the `unordered_map`.

Reply:
Having a variant which in addition to Point-cloud the indices cleared or indices kept would be desireable for adjusting application data attached to each point. Currently i do this (using TriangleMesh.remove_duplicate_vertices) by temporarily abusing the vertex/point colors as they are float values between [0,1] is just replace them by a color triplet of the kind [point_index/num_points,point_index/num_points,point_index/num_points] and than after removal simply take one color channel and multiply by num_points and round to closest integer. Having the information directly from the method would mean a great simplification.

Reply:
@reyanshsolis, My inspiration to use hashmap is originally from [`removeDuplicatedVertices` in `TriangleMesh`](https://github.com/isl-org/Open3D/blob/master/cpp/open3d/geometry/TriangleMesh.cpp#L681). I guess the point is simply that, it would be costly to find and store each unique element in `vertices_`  (and hence `colors_` into a separate array). Finding duplicated vertices otherwise would require comparing vertices (each three coordinates separately) with all other element in vertices_.

Also, using `unordered_set` to store unique vertices would not preserve the order of index creating a mess to copy `colors_` for example.

Reply:
Follow up: For legacy, I am keeping the current implementation with minor changes. Attribute re-computation must be avoided (read the last comment).

The return is just the PointCloud (no indices) (reason: i. consistent with other legacies pre-processing remove-* function, ii. as it is used in pre-processing, in a common case the indices might not be required).

The `t::PointCloud::RemoveDuplicatedPoints` may be implemented using `Tensor::Unique` op (to be added), and to be consistent with the `numpy / torch unique` op, it may return the removed / selected indices.